cmd/tailscale/cli/status.go

```diff
@@ -216,13 +216,11 @@ func peerActive(ps *ipnstate.PeerStatus) bool {
 }
 
 func dnsOrQuoteHostname(st *ipnstate.Status, ps *ipnstate.PeerStatus) string {
-	if i := strings.Index(ps.DNSName, "."); i != -1 && dnsname.HasSuffix(ps.DNSName, st.MagicDNSSuffix) {
-		return ps.DNSName[:i]
+	baseName := dnsname.ToMagicDNSBaseName(ps.DNSName, st.MagicDNSSuffix)
```

Comment:
@bradfitz I also updated `tailscale status` and `tailscale status --web` here to use the sanitation util functions. I think this should be fine so we have display names standardized everywhere, but I want to callout that this now results in dots in the hostname appearing as dashes.

In particular, `hello.ipn.dev` now looks like this:
<img width="444" alt="Screen Shot 2021-02-09 at 1 51 24 PM" src="https://user-images.githubusercontent.com/9019214/107418209-64227680-6ae4-11eb-8d72-6c1d57977f59.png">

Should be fine because this is consistent with what we display on the admin panel too:
<img width="212" alt="Screen Shot 2021-02-09 at 2 33 53 PM" src="https://user-images.githubusercontent.com/9019214/107418304-7c929100-6ae4-11eb-9009-c0830449161c.png">

But wanted to check with you before merging.

Reply:
really we'll want more discoverability in both these places. we could probably start returning a hostname from the server side.

so this is fine for now.